### PR TITLE
Wrap Google open id config in option so it can be disabled.

### PR DESCRIPTION
**Pull request by sea-snake**
Wrap Google open id config in option so it can be disabled. Additionally, change client id to struct to allow multiple properties besides client id for each open id client.

# Tests

- Updated existing test to go from enabled -> disabled

<hr/><details><summary>🟡 Some screens were changed</summary></details>

### CODE REVIEW

File: src/internet_identity/internet_identity.did [hunk 1]
```diff
@@ -237,7 +237,12 @@ type CaptchaConfig = record {
 };
 
 // Init arguments of II which can be supplied on install and upgrade.
-// Setting a value to null keeps the previous value.
+//
+// Each field is wrapped is `opt` to indicate whether the field should
+// keep the previous value or update to a new value (e.g. `null` keeps the previous value).
+//
+// Some fields, like `openid_google`, have an additional nested `opt`,
+// this indicates enable/disable status (e.g. `opt null` disables a feature).
```
Comment (lmuntaner) [inline src/internet_identity/internet_identity.did:245]: Shall we add that `null` doesn't change the feature, whereas `opt null` disabled it.

File: src/internet_identity/tests/integration/config.rs
Comment (lmuntaner) [inline src/internet_identity/tests/integration/config.rs:1]: Should we add a test that when `Some(None)` is passed it "disables" the open id? Could we use another endpoint instead of the `config` one?

File: src/internet_identity_interface/src/internet_identity/types.rs [hunk 2]
```diff
@@ -188,6 +188,13 @@ pub struct AnchorCredentials {
     pub recovery_phrases: Vec<PublicKey>,
 }
 
+/// Init arguments of II which can be supplied on install and upgrade.
+///
+/// Each field is wrapped in `Option<>` to indicate whether the field should
+/// keep the previous value or update to a new value (e.g. `None` keeps the previous value).
+///
+/// Some fields, like `openid_google`, have an additional nested `Option<>`,
+/// this indicates enable/disable status (e.g. `Some(None)` disables a feature).
```
Comment (lmuntaner) [inline src/internet_identity_interface/src/internet_identity/types.rs:197]: Ditto to say that `Some(None) disables a feature whereas None leaves it untouched`

**Review by lmuntaner — APPROVED** [review]
I understand the point, it makes sense for nicer API to disable / enable different Open ID providers.

Approving, but take a look whether you can add the test and also improve the comment.

File: src/internet_identity/internet_identity.did (hunk 1 above)
Comment (lmuntaner) [reply src/internet_identity/internet_identity.did:245]: Oh, and thanks for adding this explanation!

File: src/internet_identity/tests/integration/config.rs
Comment (sea-snake) [reply src/internet_identity/tests/integration/config.rs:1]: As discussed, this is covered in the integration test. In another PR I'll attempt to clear/split up these tests.
